posthog/models/web_preaggregated/sql.py

```diff
@@ -0,0 +1,285 @@
+):
+    team_filter = f"raw_sessions.team_id IN({team_ids})" if team_ids else "1=1"
+    person_team_filter = f"person_distinct_id_overrides.team_id IN({team_ids})" if team_ids else "1=1"
+    events_team_filter = f"e.team_id IN({team_ids})" if team_ids else "1=1"
+
+    return f"""
+    INSERT INTO {table_name}
+    SELECT
+      toStartOfDay(start_timestamp) AS day_bucket,
+      team_id,
+      host,
+      device_type,
+      uniqState(assumeNotNull(session_person_id)) AS persons_uniq_state,
+      uniqState(assumeNotNull(session_id)) AS sessions_uniq_state,
+      sumState(pageview_count) AS pageviews_count_state,
+      sumState(session_duration) AS total_session_duration_state,
+      sumState(toUInt64(ifNull(is_bounce, 0))) AS total_bounces_state
+    FROM
+      (
+        SELECT
+          any(if(NOT (empty (events__override.distinct_id)), events__override.person_id, events.person_id)) AS session_person_id,
+          events__session.session_id AS session_id,
+          e.mat_$host AS host,
+          e.mat_$device_type AS device_type,
+          any(events__session.`$session_duration`) AS session_duration,
+          any(events__session.`$is_bounce`) AS is_bounce,
+          countIf(e.event IN ('$pageview', '$screen')) AS pageview_count,
+          e.team_id AS team_id,
+          min(events__session.start_timestamp) AS start_timestamp
+        FROM events e
+        LEFT JOIN (
+          /* Session join logic */
+          SELECT
+            toString(reinterpretAsUUID(bitOr(bitShiftLeft(raw_sessions.session_id_v7, 64), bitShiftRight(raw_sessions.session_id_v7, 64)))) AS session_id,
+            min(toTimeZone(raw_sessions.min_timestamp, '{timezone}')) AS start_timestamp,
+            dateDiff('second', min(toTimeZone(raw_sessions.min_timestamp, '{timezone}')), max(toTimeZone(raw_sessions.max_timestamp, '{timezone}'))) AS `$session_duration`,
+            /* Bounce calculation logic */
+            if(ifNull(equals(uniqUpToMerge(1)(raw_sessions.page_screen_autocapture_uniq_up_to), 0), 0), NULL,
+              NOT(or(
+                ifNull(greater(uniqUpToMerge(1)(raw_sessions.page_screen_autocapture_uniq_up_to), 1), 0),
+                greaterOrEquals(dateDiff('second',
+                  min(toTimeZone(raw_sessions.min_timestamp, '{timezone}')),
+                  max(toTimeZone(raw_sessions.max_timestamp, '{timezone}'))), 10)
+              ))
+            ) AS `$is_bounce`,
+            raw_sessions.session_id_v7 AS session_id_v7
+          FROM raw_sessions
+          WHERE {team_filter}
+            AND toTimeZone(raw_sessions.min_timestamp, '{timezone}') >= toDateTime('{date_start}', '{timezone}')
+            AND toTimeZone(raw_sessions.min_timestamp, '{timezone}') < toDateTime('{date_end}', '{timezone}')
+          GROUP BY raw_sessions.session_id_v7
+          SETTINGS {settings}
+        ) AS events__session ON toUInt128(accurateCastOrNull(e.`$session_id`, 'UUID')) = events__session.session_id_v7
```

Comment:
You might be able to use `e.$session_id_uuid` here, we're just waiting for the backfill job to finish but longer term this might speed up this job quite a bit

Reply:
I didn't change this one yet as I wanted to test the whole Dagster setup, but if you think we can merge with the change, I am happy to do so. Are you running the backfill?

Reply:
Not yet
https://metabase.prod-us.posthog.dev/question/901-estimate-materialization-progress?column=%24session_id_uuid
https://metabase.prod-eu.posthog.dev/question/306-estimate-materialization-progress?column=%24session_id_uuid

More context: https://posthog.slack.com/archives/C076R4753Q8/p1743178588579349